### ISSUE TITLE
Revert the style2str change in b547ead

### ISSUE THREAD

**Issue by phadej** [haskell/cabal#8721]
The commit b547ead58bf09bb838c13f02afb2f1042ad1bc7c introduced change to `style2str`. I think it's wrong. Or at least I'd want to see a lot better justification for it then just changing inside unrelated patch.

In particular, there are no tests. As a maintainer of `cabal-plan` I want to understand the change, but I don't. So please revert it, and do it properly if there is justification.

**Comment by jneira** [comment]
Just in case, from the commit message:

> Additionally, adjust the ‘style’ attribute in plan.json so that globally
installed packages are designated as global even if they are local to the
project.  (Without this adjustment to ‘style2str’, the T5782Diamond test fails,
because it looks up ‘dist-dirs’ in plan.json, where ‘dist-dirs’ is absent from
the JSON.)

But there is no specific tests about that change (but it did not broke any prior test so the previous behaviour was not tested either?)

**Comment by Mikolaj** [comment]
@jneira: yes, it seems it wasn't tested. Which just proves you right when you go around pestering everybody to add their tests to their PRs. :D

**Comment by phadej** [comment]
> But there is no specific tests about that change (but it did not broke any prior test so the previous behaviour was not tested either?)

There kind of was. The beef of b547ead58bf09bb838c13f02afb2f1042ad1bc7c broke `T5782Diamond` test. So instead of thinking that changing what local package means is maybe not the right way, the author made another somewhat unrelated change elsewhere.

It isn't a direct test about what `plan.json`s `local`, `inplace` and `global` means, but I explain my best understanding in https://github.com/haskell/cabal/pull/8722#issuecomment-1412590671. The b547ead58bf09bb838c13f02afb2f1042ad1bc7c introduced the fourth combination in `ElabConfiguredPackage` flags , which I think shouldn't ever happen, as it doesn't make sense to me. OTOH, it's not prevented by the types, so it does happen occasinonally.

I'd go and try to change the two booleans into a type with only three values, and see what changes need to be done. But it's a time consuming experiment, so I won't.
